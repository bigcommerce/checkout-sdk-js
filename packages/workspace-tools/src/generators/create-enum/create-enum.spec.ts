import path from 'path';

import createEnum from './create-enum';

describe('createEnum()', () => {
    it('export matching members from files to another file', async () => {
        const options = {
            inputPaths: [path.join(__dirname, '/__fixtures__/**/index.ts')],
            inputMemberPattern: '^Type.+',
            outputPath: path.join(__dirname, '/__temp__/output.ts'),
            outputMemberName: 'Type',
        };

        expect(await createEnum(options)).toMatchSnapshot();
    });

    it('handles scenario where no matching member is found', async () => {
        const options = {
            inputPaths: [path.join(__dirname, '/__fixtures__/**/index.ts')],
            inputMemberPattern: '^Test',
            outputPath: path.join(__dirname, '/__temp__/output.ts'),
            outputMemberName: 'Type',
        };

        expect(await createEnum(options)).toMatchSnapshot();
    });
});
