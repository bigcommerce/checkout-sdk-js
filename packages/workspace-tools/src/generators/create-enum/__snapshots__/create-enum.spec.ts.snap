// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createEnum() export matching members from files to another file 1`] = `
"enum Type {
    TYPEA = \\"typeA\\",
    TYPEB = \\"typeB\\"
}
export default Type;
"
`;

exports[`createEnum() handles scenario where no matching member is found 1`] = `
"enum Type {
}
export default Type;
"
`;
